internal/envoy/v3/route_test.go

```diff
@@ -2728,3 +2727,75 @@ func TestRouteRedirect(t *testing.T) {
 }
 
 func virtualhosts(v ...*envoy_route_v3.VirtualHost) []*envoy_route_v3.VirtualHost { return v }
+
+func TestVirtualHostAndRoutes(t *testing.T) {
+	tests := map[string]struct {
+		vh     *dag.VirtualHost
+		route  []*dag.Route
+		secure bool
+		want   *envoy_route_v3.VirtualHost
+	}{
+		"dag virtual host to envoy virtual host with custom header": {
+			vh: &dag.VirtualHost{
+				Name: "example.com",
+				RequestHeadersPolicy: &dag.HeadersPolicy{
+					Set: map[string]string{
```

Comment:
We're only testing `Set` and `Remove` here though the implementation also includes supporting the `Add` operation for headers. Technically the HTTPProxy API does not have an `Add` mechanism so we could either omit it in the implementation or just leave it but make sure it is tested here

Reply:
Added test cases for Add case also